mono/utils/mono-threads-state-machine.c

```diff
@@ -539,13 +645,15 @@ mono_thread_info_get_suspend_state (MonoThreadInfo *info)
 	case STATE_SELF_SUSPENDED:
 	case STATE_BLOCKING_AND_SUSPENDED:
 		return &info->thread_saved_state [SELF_SUSPEND_STATE_INDEX];
+	case STATE_BLOCKING_ASYNC_SUSPENDED:
```

Comment:
You can just add this case at https://github.com/mono/mono/pull/8068/files#diff-6efe503cfde87f09553ab6c97c061604R644